### DISCUSSION TITLE
Add scope to "imported" documents

### DISCUSSION THREAD

**Discussion by luiz290788** [eclipse-langium/langium#979] [Q&A] [answered]
I'm working in a solution with multiple languages one that defines a model and the other that builds on top of multiple instances of that model. But different instances of the base model might have the same id for the same entity, is there any way to scope that in the model that uses the base?

Let me try to give an example.

Let's say we have a model for citizens of countries:

```langium
grammar Country

entry Model: (citizens+=Citizen)*;

Citizen:
    'citizen' name=ID;

hidden terminal WS: /\s+/;
terminal ID: /[_a-zA-Z][\w_]*/;

hidden terminal ML_COMMENT: /\/\*[\s\S]*?\*\//;
hidden terminal SL_COMMENT: /\/\/[^\n\r]*/;
```

And then I can have multiple countries:

Brazil.country
```country
citizen Ronaldo
```

Portugal.country
```country
citizen Ronaldo
```

And then I have a model for a football team:

```langium
grammar FootballTeam

import "./citizen"

entry Team: (citizens+=Players)*;

Players: 'player' player=[Citizen:ID];
```

Where I want to declare a team like this:

```team
player Ronaldo // here I want to reference the Ronaldo from Brazil
```

If I have all these files in the same folder and load all of them in the language server, how can I be specific to which `Ronaldo` I am referencing?

I almost want the import feature from typescript that I can name a import and references objects inside of it, is it possible?

**Comment by msujew** [comment, marked as answer]
Hey @luiz290788,

you would need some sort of import mechanism for that. We actually use that in our own [grammar-scope](https://github.com/langium/langium/blob/main/packages/langium/src/grammar/references/grammar-scope.ts#L54-L66). You need to restrict what you import from the global scope by looking at the URIs of the `AstNodeDescriptions`.

**Reply by luiz290788** [reply]
This is great! I was able to implement an example using the scope provider. [Here](https://codesandbox.io/p/sandbox/football-m8pk80) it is in case someone is looking for something similar.

I will continue evolving that to work with auto completes import the player files dynamically but I was able to achieve what I wanted.
